macos/Sources/Features/Terminal/TerminalController.swift

```diff
@@ -268,6 +273,11 @@ class TerminalController: BaseTerminalController {
             // so we only call this if we are opaque.
             window.updateTabBar()
         }
+
+        guard let windows = self.window?.tabbedWindows as? [TerminalWindow] else { return }
+        for (w) in windows {
+            w.titleForegroundColor = window.titleForegroundColor
+        }
```

Comment:
It feels like this should be moved to a private func and this should be called here and in `windowDidBecomeKey` instead of a _full `syncAppearance`_ which is rather expensive. Thoughts?

Reply:
That sounds good and reasonable to me.  I'll make that change later today and do some more manual testing to make sure all the existing cases are still handled.

Reply:
Updated in https://github.com/ghostty-org/ghostty/pull/3843/commits/3c17b9176e6c6cbfca6b83ce5943c5b185480ebb